### ISSUE TITLE
[dynamodb] global secondary index is not created/deleted when applying new changes 

### ISSUE THREAD

**Issue by Julian-Chu** [aws-controllers-k8s/community#1596]
**Describe the bug**
the global secondary index can't be created or deleted with applying new changes

**Steps to reproduce**
create table without global secondary index
```yaml
apiVersion: dynamodb.services.k8s.aws/v1alpha1
kind: Table
metadata:
  name: ack-demo-table-provisioned-global-secondary-indexes
spec:
  attributeDefinitions:
    - attributeName: Id
      attributeType: S
  keySchema:
    - attributeName: Id
      keyType: HASH
#  globalSecondaryIndexes:
#    - indexName: id-index
#      keySchema:
#        - attributeName: Id
#          keyType: HASH
#      projection:
#        projectionType: ALL
#      provisionedThroughput:
#        readCapacityUnits: 1
#        writeCapacityUnits: 1

  provisionedThroughput:
      readCapacityUnits: 1
      writeCapacityUnits: 1
  billingMode: "PROVISIONED"
  tableName: "ack-demo-table-provisioned-global-secondary-indexes"
```

add global secondary index to table
```yaml
apiVersion: dynamodb.services.k8s.aws/v1alpha1
kind: Table
metadata:
  name: ack-demo-table-provisioned-global-secondary-indexes
spec:
  attributeDefinitions:
    - attributeName: Id
      attributeType: S
  keySchema:
    - attributeName: Id
      keyType: HASH
  globalSecondaryIndexes:
    - indexName: id-index
      keySchema:
        - attributeName: Id
          keyType: HASH
      projection:
        projectionType: ALL
      provisionedThroughput:
        readCapacityUnits: 1
        writeCapacityUnits: 1

  provisionedThroughput:
      readCapacityUnits: 1
      writeCapacityUnits: 1
  billingMode: "PROVISIONED"
  tableName: "ack-demo-table-provisioned-global-secondary-indexes"
```

no global secondary index created, same behaviour by deletion
**Expected outcome**
global secondary index should be created/deleted.

**Environment**

Kubernetes version: 1.23
EKS: 1.23
AWS service targeted: DynamoDB
DynamoDB controller: 0.1.7

**Comment by jrake-revelant** [comment]
@A-Hilaly why is this considered a feature request?

**Comment by A-Hilaly** [comment]
Hi @jrake-revelant , i'm aware that the update code path for Table resources isn't fully complete and i'm actively working on the remaining features :)

**Comment by jljaco** [comment]
To be addressed by https://github.com/aws-controllers-k8s/dynamodb-controller/pull/30

**Comment by A-Hilaly** [comment]
Fixed and released in dynamodb `v1.1.0`
/close
